File: rules/pascal/tests/pas.sql-injection/sql.pas
program SqlTests;
var q: String;
begin
  q := 'SELECT * FROM t WHERE id=' + ParamStr(1);
  Query1.SQL.Text := q; // crit:expect pas.sql-injection
  ExecuteDirect('SELECT * FROM t WHERE x=' + ParamStr(2)); // crit:expect pas.sql-injection
  Query2.SQL.Text := 'SELECT * FROM t WHERE active=1'; // crit:expect-not pas.sql-injection
  ExecuteDirect('SELECT 1'); // crit:expect-not pas.sql-injection
end.

File: rules/rust/tests/rs.sql-injection.format/sql.rs
fn queries(conn: &Conn, id: i64) {
    sqlx::query(&format!("SELECT * FROM u WHERE id = {}", id)); // crit:expect rs.sql-injection.format
    conn.execute(format!("DELETE FROM u WHERE id = {}", id)); // crit:expect rs.sql-injection.format
    sqlx::query("SELECT * FROM u WHERE id = $1"); // crit:expect-not rs.sql-injection.format
    conn.execute("DELETE FROM u WHERE active = 0"); // crit:expect-not rs.sql-injection.format
}
